apps/openmw/mwmechanics/actors.cpp

```diff
@@ -557,28 +556,25 @@ namespace MWMechanics
         stats.setFatigue (fatigue);
     }
 
-    class ExpiryVisitor : public EffectSourceVisitor
+    class SortByRemainingTimeVisitor : public EffectSourceVisitor
     {
-        private:
-            MWWorld::Ptr mActor;
-            float mDuration;
+            private:
+            bool unused;
 
-        public:
-            ExpiryVisitor(const MWWorld::Ptr& actor, float duration)
-                : mActor(actor), mDuration(duration)
+            public:
+            std::vector<std::tuple<float, float, MWMechanics::EffectKey>> mEffectsByRemainingTime;
```

Comment:
Do you have a suggestion? I wanted to pull those three pieces of information out from the visitor class.